pyvista/core/filters/image_data.py

```diff
@@ -946,3 +949,134 @@ def contour_labeled(
         # Restore the original vtkLogger verbosity level
         _vtk.vtkLogger.SetStderrVerbosity(verbosity)
         return cast(pyvista.PolyData, wrap(alg.GetOutput()))
+
+    def pad_image(
+        self,
+        value: float = 0.0,
+        *,
+        width: int | VectorLike[int] = 1,
+        all_dimensions: bool = False,
+        scalars: Optional[str] = None,
+        all_point_data: bool = True,
+        progress_bar=False,
+    ) -> pyvista.ImageData:
+        """Pad an image.
+
+        Pad image point data with a constant value.
+
+        Parameters
+        ----------
+        value : int | float, default : 0
+            Constant value to use for padded elements.
+
+        width : int | sequence[int], default : 1
+            Amount of padding to apply to each axis. This value represents the number of
+            points to add to the extents of the image. Specify:
+
+            - A single value to apply constant padding around the entire image.
+            - Three values, one for each ``(X, Y, Z)`` axis, to apply symmetrical
+              padding to each axis independently.
+            - Six values, one for each ``(-X,+X,-Y,+Y,-Z,+Z)`` direction, to apply
+              padding to each direction independently.
+
+        all_dimensions : bool, default : False
+            Control whether to pad all dimensions. If ``True``, all dimensions are
+            padded. If ``False`` (default), only non-singleton dimensions are padded.
+            This means that 2D images (which have a third, singleton dimension) will
+            remain 2D after padding by default.
+
+        scalars : str, optional
+            Name of scalars to pad. Defaults to currently active scalars. All other
+            data arrays are ignored are not be included in the output. Has no effect
+            when ``all_point_data`` is ``True``.
+
+        all_point_data : bool, default: True
+            Pad all point data arrays and include them in the output. If ``False``,
+            only the specified ``scalars`` are padded.
+
+        progress_bar : bool, default: False
+            Display a progress bar to indicate progress.
+
+        Returns
+        -------
+        pyvista.ImageData
+            Padded image.
```

Comment:
Maybe this will be added soon, but please add a section in Examples to visually teach the use of this filter:)

Reply:
Yes I will add a few examples soon

Reply:
Examples were added in https://github.com/pyvista/pyvista/pull/6051/commits/5cdcb4495f346ce38c0f42ddf34aa77ea416ae7a